arch/arc/core/thread.c

```diff
@@ -194,12 +194,21 @@ void arch_new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 	/* initial values in all other regs/k_thread entries are irrelevant */
 }
 
+#ifdef CONFIG_MULTITHREADING
 void *z_arch_get_next_switch_handle(struct k_thread **old_thread)
 {
 	*old_thread =  _current;
 
 	return z_get_next_switch_handle(NULL);
 }
+#else
+void *z_arch_get_next_switch_handle(struct k_thread **old_thread)
```

Comment:
Agree here. Indeed I plan to do this improvement, though I plan to do it separately as a part of ARC interrupt code overhaul which I'm working on.

It won't give big improvement in either image size or execution speed, but without additional overhaul it will bring lot's of additional ifdefs checking for CONFIG_MULTITHREADING in lot's of places in ASM ARC code. So I'd prefer to do it clear on separate step.